### PR TITLE
Webpack 2.1.0-beta.25

### PR DESCRIPTION
**Pull request by peter-mouland**
> fixes https://github.com/mxstbr/react-boilerplate/issues/1027
- ~~renamed webpack._.babel.js to webapack._.js as the .babel postfix doesn't work in webpack 2 and it doesnt seem to be used in this project anyway.~~
- ~~removed the sourcemap query from dev as webpack 2 automatically passes through to plugins when set as source-map in the dev-tool option~~
- removed prod jsUglify as this is automatically done with `-p` arg
- moved postCSS config to LoaderOptionsPlugin within webpack

### CODE REVIEW

**Comment by mxstbr** [comment]
Deploy preview failed.

Built with commit 57794f48410d3b006f0afa72d14fa5973c9ea472

https://app.netlify.com/sites/react-boilerplate/deploys/57e7927aa7e2d231628e6a1d

**Comment by peter-mouland** [comment]
I accidentally did the PR against master to start with and had to change to it dev soon after, could that be it?

I can't see anything after logging into that link. hat tab should i be looking at?

**Comment by peter-mouland** [comment]
the previous fail was due to https://github.com/mxstbr/react-boilerplate/pull/1031

**Comment by ansumanshah** [comment]
Tried to check this but css-modules not scoping locally with this patch. Any idea what might be the issue?

**Comment by peter-mouland** [comment]
cool, thanks for taking a look @ansumanshah, could you explain what you mean?  Are the class names as you'd expect or is it the styling not being applied correctly?

I'll take a look to see if i see the same thing my end...

**Comment by alexanbj** [comment]
You have to add `context: __dirname` to the LoaderOptionsPlugin. See https://github.com/webpack/webpack/issues/3018#issuecomment-248633498

**Comment by peter-mouland** [comment]
i've added that but i'm still not sure what was wrong about the output, or what you were seeing that I am not?

**Comment by alexanbj** [comment]
As @ansumanshah said, CSS module classnames were no longer locally scoped.

Not sure if this project has any identical classnames, but in my own project, which is based on this boilerplate and upgraded to webpack 2.1.0-beta.25 by looking at this pull request (thanks by the way!), I suddenly got classname crashes. Adding context seems to fix it though.

File: docs/general/faq.md [hunk 1]
```diff
@@ -141,7 +141,7 @@ WebStorm is a powerful IDE, and why not also use it as debugger tool? Here is th
 
 1.  [Install JetBrain Chrome Extension](https://chrome.google.com/webstore/detail/jetbrains-ide-support/hmhgeddbohgjknpmjagkdomcpobmllji)
 2.  [Setting up the PORT](https://www.jetbrains.com/help/webstorm/2016.1/using-jetbrains-chrome-extension.html)
-3.  Change WebPack devtool config to `source-map` [(This line)](https://github.com/mxstbr/react-boilerplate/blob/56eb5a0ec4aa691169ef427f3a0122fde5a5aa24/internals/webpack/webpack.dev.babel.js#L65)
+3.  Change WebPack devtool config to `source-map` [(This line)](https://github.com/mxstbr/react-boilerplate/blob/56eb5a0ec4aa691169ef427f3a0122fde5a5aa24/internals/webpack/webpack.dev.js#L65)
```
Comment (mxstbr) [inline docs/general/faq.md]: This will break because it links to a _specific commit_. At this commit, the file is (and always will be) called `webpack.dev.babel.js`. Please update the entire link to the latest commit on `master`, linking to the exact same code!

File: internals/webpack/webpack.base.js [hunk 2]
```diff
@@ -70,20 +70,24 @@ module.exports = (options) => ({
...
     }),
-  ]),
-  postcss: () => [
-    postcssFocus(), // Add a :focus to every :hover
-    cssnext({ // Allow future CSS features to be used, also auto-prefixes the CSS...
-      browsers: ['last 2 versions', 'IE > 10'], // ...based on this browser list
-    }),
-    postcssReporter({ // Posts messages from plugins to the terminal
-      clearMessages: true,
+    new webpack.LoaderOptionsPlugin({
```
Comment (mxstbr) [inline internals/webpack/webpack.base.js]: Add comment above explaining what this does. Something like:

```JS
// Pass options to loaders
```

File: internals/webpack/webpack.base.js [hunk 3]
```diff
@@ -70,20 +70,24 @@ module.exports = (options) => ({
...
+      options: {
+        context: __dirname,
+        postcss: [
```
Comment (mxstbr) [inline internals/webpack/webpack.base.js]: Add comment above, saying something like

```JS
// Configure PostCSS loader
```

File: internals/webpack/webpack.base.js [hunk 4]
```diff
@@ -70,20 +70,24 @@ module.exports = (options) => ({
...
-  postcss: () => [
-    postcssFocus(), // Add a :focus to every :hover
-    cssnext({ // Allow future CSS features to be used, also auto-prefixes the CSS...
-      browsers: ['last 2 versions', 'IE > 10'], // ...based on this browser list
-    }),
-    postcssReporter({ // Posts messages from plugins to the terminal
-      clearMessages: true,
+    new webpack.LoaderOptionsPlugin({
+      options: {
+        context: __dirname,
```
Comment (mxstbr) [inline internals/webpack/webpack.base.js]: Add comment above saying this is needed for `css-loader` and linking to the relevant issue in the webpack repo.

File: internals/webpack/webpack.base.js [hunk 5]
```diff
@@ -96,5 +100,4 @@ module.exports = (options) => ({
   devtool: options.devtool,
   target: 'web', // Make web variables accessible to webpack, e.g. window
   stats: false, // Don't show stats in the console
-  progress: true,
```
Comment (mxstbr) [inline internals/webpack/webpack.base.js]: Why was this moved to the `package.json`? What's the benefit?

File: internals/webpack/webpack.prod.js [hunk 6]
```diff
@@ -39,13 +39,6 @@ module.exports = require('./webpack.base.babel')({
     // Merge all duplicate modules
     new webpack.optimize.DedupePlugin(),
 
-    // Minify and optimize the JavaScript
-    new webpack.optimize.UglifyJsPlugin({
```
Comment (mxstbr) [inline internals/webpack/webpack.prod.js]: Why was this removed?

File: package.json [hunk 7]
```diff
@@ -105,7 +105,10 @@
       }
     },
     "rules": {
-      "arrow-parens": ["error", "always"],
+      "arrow-parens": [
+        "error",
+        "always"
+      ],
```
Comment (mxstbr) [inline package.json]: Please undo whitespace changes here.

File: package.json [hunk 8]
```diff
@@ -141,7 +144,10 @@
       "prefer-template": 2,
       "class-methods-use-this": 0,
       "react/forbid-prop-types": 0,
-      "react/jsx-first-prop-new-line": [2, "multiline"],
+      "react/jsx-first-prop-new-line": [
+        2,
+        "multiline"
+      ],
```
Comment (mxstbr) [inline package.json]: Please undo whitespace changes here.

File: internals/webpack/webpack.prod.js (hunk 6 above)
Comment (mxstbr) [inline internals/webpack/webpack.prod.js]: Sorry, ignore, didn't read OP. "removed prod jsUglify as this is automatically done with -p arg"

File: internals/webpack/webpack.base.js (hunk 5 above)
Comment (peter-mouland) [inline internals/webpack/webpack.base.js]: progress is now an invalid property to be passed in the config. webpack fails if it was still present:

```sh
Invalid configuration object. Webpack has been initialised using a configuration object that does not match the API schema.
 - configuration has an unknown property 'progress'. These properties are valid:
   object { amd?, bail?, cache?, context?, dependencies?, devServer?, devtool?, entry, externals?, loader?, module?, name?, node?, output?, plugins?, profile?, recordsInputPath?, recordsOutputPath?, recordsPath?, resolve?, resolveLoader?, stats?, target?, watch?, watchOptions? }

```

File: package.json (hunk 7 above)
Comment (peter-mouland) [inline package.json]: Whoops, these came from 3.2.1 (master).  i accidentally rebased from master instead of dev.  

Shouldn't dev be rebase from master too?

File: docs/general/faq.md (hunk 1 above)
Comment (peter-mouland) [inline docs/general/faq.md]: done

File: internals/webpack/webpack.base.js (hunk 2 above)
Comment (peter-mouland) [inline internals/webpack/webpack.base.js]: no probs - updated

File: internals/webpack/webpack.base.js (hunk 4 above)
Comment (peter-mouland) [inline internals/webpack/webpack.base.js]: updated..

File: internals/webpack/webpack.base.js (hunk 3 above)
Comment (peter-mouland) [inline internals/webpack/webpack.base.js]: added

**Comment by peter-mouland** [comment]
@mxstbr as I mentioned in the first pr comments, the `.babel` in webpack config no longer works for webpack 2.  here is the error when using es6 goodies that babel should translate:

```sh
/Users/peterm/Workspace/forks/react-boilerplate/internals/webpack/webpack.prod.babel.js:28
  cssLoaders: ExtractTextPlugin.extract({ ...opts }),
                                          ^^^
SyntaxError: Unexpected token ...

```

File: package.json (hunk 7 above)
Comment (mxstbr) [inline package.json]: Oh if these are in master I missed them somewhere 😊 you can leave them in in that case

File: internals/webpack/webpack.base.js (hunk 5 above)
Comment (mxstbr) [inline internals/webpack/webpack.base.js]: Woah that's weird. @TheLarkinn why was this removed, or is this a bug in the validator? /cc @gajus

**Comment by mxstbr** [comment]
@peter-mouland is there a source on why that was removed? Seems like a weird omission to just make. /cc @TheLarkinn

**Comment by peter-mouland** [comment]
not that i found - but I also haven't seen the `.babel.js` postfix in the official docs in the first place either!  I probably  missed it, but it seemed to me like _secret_ functionality and a bit _magic_. In that I mean, you don't tell babel to look at webpack.conf.babel.js and you haven't updated the webpack config to tell it to also pull in babel first.  it was all a bit clandestine!

**Comment by peter-mouland** [comment]
according to this thread (https://github.com/webpack/webpack/issues/1403#issuecomment-244339631) people do seem to have it working, not that i could when I tried quickly.  

How does this boilerplate read the babel config, do all builds use the config set in the package.json automatically?

**Comment by salzhrani** [comment]
Is `npm run analyze` working properly?

**Comment by mxstbr** [comment]
Is this ready to be reviewed against Peter? Would love to land this update!

**Comment by peter-mouland** [comment]
i've rebased against dev again.  

The only outstanding issue is the removal of the `.babel.js` postfix which i wasn't able to get working anyway.

analyse seems to work...
screen shot 2016-10-17 at 21 46 56

**Comment by peter-mouland** [comment]
is the .babel thing a must? i.e. do you see it asa  feature of this project to allow people to change the webpack files in es6 even though none of those are used by default?

**Comment by peter-mouland** [comment]
@gihrig @mxstbr I've rebased this branch from dev and removed anything to do with changing the file names.

It should be good to go now.

**Comment by gihrig** [comment]
I have run the battery of tests I use on dependency update - all pass 🎉 

@mxstbr LGTM! Just need to release your review, and convince Coveralls to complete...

**Comment by mxstbr** [comment]
Awesome, thanks @peter-mouland!
